spec/middleware/sync-storage-manager-spec.js

```diff
@@ -48,4 +50,31 @@ describe('sync-brainstem middleware', () => {
 
     expect(this.posts.get(76)).not.toBeDefined();
   });
+
+  describe('when state.brainstem has changed', () => {
+    it('syncs with the storageManager', function () {
+      spyOn(this.storageManager, 'storage').and.callThrough();
+      this.store.dispatch({
+        type: 'REMOVE_MODEL',
+        payload: {
+          brainstemKey: 'posts',
+          attributes: { id: 76 },
+        },
+      });
+
+      const actualArgs = extractArgsFromCalls(this.storageManager.storage);
```

Comment:
Is this the pattern for testing syncing? It seems way too shallow by banking on calling an internal storage manager reference. 

Unless this is the pattern, I would appreciate an end-to-end test. This will facilitate future refactorings.

Reply:
I'm not familiar with a specific pattern for testing syncing. Ideally we could see if syncBrainstem was called or not in the middleware or not, but I wasn't sure how to make that happen, so I opted for testing that something in that function was called or not. 

Also, in this case, I'm not sure what a proper end-to-end test would look like, so please let me know if you have something specific in mind.

Reply:
@juanca, this is an odd question, but what do you mean by that thumbs up? I'm not sure if it means I can just keep the tests or what.